src/bot/core/music.py

```diff
@@ -0,0 +1,292 @@
+import asyncio
+import json
+import time
+
+import discord
+import requests
+import youtube_dl
+from bs4 import BeautifulSoup
+from discord.ext import commands
+
+from .utils import listener
+from .utils.MGCert import Level, MGCertificate
+from .utils.MsgFormat import MsgFormatter
+
+song_list = list()
+
+
+class Song():
+    def addSong(self, title, url):
+        self.url = url
+        self.title = title
+
+    def searchSong(self, content, number):
+        self.url = 'https://www.youtube.com/watch?v=' + \
+            content[number]['videoRenderer']['videoId']
+        self.title = ''
+        for i in content[number]['videoRenderer']['title']['runs']:
+            self.title += i['text']
+        self.channel = content[number]['videoRenderer']['longBylineText']['runs'][0]['text']
+        self.published_time = content[number]['videoRenderer']['publishedTimeText']['simpleText']
+        self.viewCount = content[number]['videoRenderer']['viewCountText']['simpleText']
+        self.length = content[number]['videoRenderer']['lengthText']['simpleText']
+
+    def printSong(self):
+        # markdown?
+        print()
+        return
+
+
+class Music(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self._last_member = None
+        self.songQueue = 0
+
+    async def player(self, ctx: commands.Context):
+        if ctx.voice_client == None:
+            if ctx.author.voice:
+                await ctx.author.voice.channel.connect()
+                await self.playMusic(ctx)
+            else:
+                await ctx.send(embed=MsgFormatter.get(ctx, "Usage Error", "You are not in any voice channel. Please join a voice channel to use Music bot."))
+                raise commands.CommandError(
+                    "Author not connected to a voice channel.")
+
+    async def playMusic(self, ctx):
+        def next():
+            if len(song_list) > self.songQueue:
+                self.songQueue += 1
+                fut = asyncio.run_coroutine_threadsafe(
+                    self.playMusic(ctx), self.bot.loop)
+                try:
+                    fut.result()
+                except:
+                    pass
+
+        await ctx.send(embed=MsgFormatter.get(ctx, "Now Playing",
+                                              song_list[self.songQueue].title + "  " + song_list[self.songQueue].length))
+
+        ydl_opts = {
+            'format': 'bestaudio/best',
+            'postprocessors': [{
+                'key': 'FFmpegExtractAudio',
+                'preferredcodec': 'mp3',
+                'preferredquality': '192',
+            }]
+        }
+        with youtube_dl.YoutubeDL(ydl_opts) as ydl:
+            info = ydl.extract_info(
+                song_list[self.songQueue].url, download=False)
```

Comment:
```suggestion
           loop = asyncio.get_event_loop()
           data = await loop.run_in_executor(None, lambda: ytdl.extract_info(song_list[self.songQueue].url, download=False))
```
Consider running in an asyncio event loop. Functions that involve network communication like this must be executed asynchronously.